File: Questão-2/Carro.go
package main

import (
	"fmt"
)
//Classe Carro Criada
type Carro struct{
	marca string
	modelo string
	ano int
	velocidade int
}

//Método InformaCarro, que retorna os dados do carro
func(c Carro)InformaCarro(){
	fmt.Printf("Marca: %s Modelo: %s Ano: %d\n\n", c.marca, c.modelo, c.ano)
}

//Método Acelera, que aumenta a velocidade do carro
func(c *Carro)Acelerar(){
	c.velocidade += 10
	fmt.Printf("Acelerando para %d km/h\n\n", c.velocidade)
}
//Método Freia, que diminui a velocidade do carro
func(c *Carro)Frear(){
	c.velocidade -= 10
	fmt.Printf("Freando para %d km/h\n\n", c.velocidade)
}

//Método Velocidade, que exibe a velocidade do carro
func(c Carro)MostraVelocidade(){
	fmt.Printf("Velocidade: %dkm/h\n\n", c.velocidade)
}

func main() {
//Objetos da classe carro criados
carro1 := Carro{marca: "Fiat", modelo: "Uno", ano: 2011, velocidade: 0}

//Exibe Detalhes
carro1.InformaCarro()
//Acelera
carro1.Acelerar()
carro1.Acelerar()
//Freia
carro1.Frear()	
//Mostra Velocidade
carro1.MostraVelocidade()

File: Questão-2/Carro.py
class Carro:
  def __init__(self, marca, modelo, ano, velocidade):
    self.marca = marca
    self.modelo = modelo
    self.ano = ano
    self.velocidade = velocidade

  def InformaCarro(self):
    print(f'Marca: {self.marca} Modelo: {self.modelo} Ano: {self.ano}\n')

  def Acelerar(self):
    self.velocidade += 10
    print(f'Acelerando para {self.velocidade}Km/h')

  def Frear(self):
    self.velocidade -= 10
    print(f"Freando para {self.velocidade}Km/h")

  def MostrarVelocidade(self):
    print(f"Velocidade atual: {self.velocidade}Km/h")




carro1 = Carro("Fiat", "Uno", 2011, 0)

carro1.InformaCarro()

carro1.Acelerar()
carro1.Acelerar()
carro1.Frear()
carro1.MostrarVelocidade()
